### PR TITLE
[feat]: [IAC-142]: Add support for the properties.iacm.codebase block

### PR DESCRIPTION
**Pull request by nassergonzalez**
## Describe your changes
This PR changes the code so there can be now 2 codebase blocks, the CI and the IACM block.

As an example:

```
  properties:
    ci:
      codebase:
        connectorRef: test
        repoName: cf_templates_test
        build:
          spec:
            branch: cf_templates_test
          type: branch
    iacm:
      codebase:
        connectorRef: foobar
        repoName: infra-test
        build:
          spec:
            branch: sqs
          type: branch
```
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42403)

### CODE REVIEW

File: 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java [hunk 1]
```diff
@@ -508,11 +508,11 @@ private CodeBase getCICodebase(PlanCreationContext ctx) {
     CodeBase ciCodeBase = null;
     try {
       YamlNode properties = YamlUtils.getGivenYamlNodeFromParentPath(ctx.getCurrentField().getNode(), PROPERTIES);
-      YamlNode ciCodeBaseNode = properties.getField(CI).getNode().getField(CI_CODE_BASE).getNode();
+      YamlNode ciCodeBaseNode = properties.getField(IACM).getNode().getField(CI_CODE_BASE).getNode();
```
Comment (scottyw) [inline 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java:511]: The field name `ciCodeBaseNode` is slightly misleading now.

Is `CI_CODE_BASE` just the value `codebase`?

File: 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java [hunk 2]
```diff
@@ -508,11 +508,11 @@ private CodeBase getCICodebase(PlanCreationContext ctx) {
...
       ciCodeBase = IntegrationStageUtils.getCiCodeBase(ciCodeBaseNode);
     } catch (Exception ex) {
       // Ignore exception because code base is not mandatory in case git clone is false
```
Comment (scottyw) [inline 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java:514]: I think that the "code base is not mandatory" for CI but it is now mandatory for us. We maybe need to fail when this doesn't work.

File: 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java (hunk 1 above)
Comment (nassergonzalez) [reply 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java]: Yeah sorry. I renamed everything to iacmCodeBase to make it clearer

File: 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java (hunk 2 above)
Comment (nassergonzalez) [reply 310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java:514]: Yep, makes sense, I added a verification in the filter so if the IACM stage is there then the IACM code base needs to be present and have a connector.

**Comment by nassergonzalez** [comment]
trigger messagecheck

**Review by scottyw — APPROVED** [review]

**Review by scottyw — APPROVED** [review]
> Reviewed 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42403)* status: 2. unresolved discussions.

**Review by scottyw — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42403)* status: :shipit: complete! ✓

___
*[`310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java` line 511 at r1](https://reviewable.io/reviews/harness/harness-core/42403#-NLMhXWQHtH01Ex59awo-r1-511:-NLQxI2OCcyYQhqIcoCu:b-85i64v) ([raw file](https://github.com/harness/harness-core/blob/805985875a89ba6e2e14363d81fc10628eb3b76f/310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java#L511)):*
<details><summary><i>Previously, nassergonzalez (Nasser Gonzalez Gonzalez) wrote…</i></summary><blockquote>

Yeah sorry. I renamed everything to iacmCodeBase to make it clearer
</blockquote></details>

LGTM
___
*[`310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java` line 514 at r1](https://reviewable.io/reviews/harness/harness-core/42403#-NLMhXWQHtH01Ex59awo-r1-514:-NLQxT9a2zR71gcHQ8ET:bb6ywmx) ([raw file](https://github.com/harness/harness-core/blob/805985875a89ba6e2e14363d81fc10628eb3b76f/310-iacm-manager/app/src/main/java/io/harness/iacm/plan/creator/stage/IACMStagePMSPlanCreator.java#L514)):*
<details><summary><i>Previously, nassergonzalez (Nasser Gonzalez Gonzalez) wrote…</i></summary><blockquote>

Yep, makes sense, I added a verification in the filter so if the IACM stage is there then the IACM code base needs to be present and have a connector.
</blockquote></details>

LGTM

**Pull request merged**